src/adjustxposfunctor.cpp

```diff
@@ -246,6 +246,15 @@ FunctorCode AdjustXPosFunctor::VisitMeasure(Measure *measure)
 
         m_measureTieEndpoints = measure->GetInternalTieEndpoints();
         measure->m_measureAligner.Process(*this);
+
+        // In some cases with low enough linear spacing xRel of barline might shift due to overlaps with previous
```

Comment:
Do you understand why the barline shifts? The fix works, but I am not sure I understand its logic. Isn't there somewhere a place to fix the issue more at its root?

Reply:
Yes. The issue here is that positioning of timestamp-only alignments is done by placing them proportionally to their time. However as this happens before any of the overlaps can be processed, this can lead to situations where timestamp alignment initially matches that of non-timestamp (in current case barline) but after overlaps are calculated actual position of the timstamp remains unchanged (since the element itself does not have any overlaps). This does not generally happen unless linear spacing is low enough, like in current example.
To illustrate it further, here is the same example (without the fix), but there is no overlap happening (since only stem is present):
![image](https://user-images.githubusercontent.com/1819669/231731301-29c162f3-9b8c-425a-877f-69ea9867db4b.png)
And this is example with original problem:
![image](https://user-images.githubusercontent.com/1819669/231731478-af195da7-ec56-4e60-9809-d0859deb00df.png)
As you can see, position of timestamps remain unchanged.

The problem here is that barline and fermata don't have the same alignments, so the shift for one doesn't translate into shift for another. Alternatively this could be fixed by merging them, but I'm not sure whether them being in BARLINE alignment instead of DEFAULT could lead to some issues.

Reply:
Thank you for the explanation. I am still looking into it.